tests/linearizability/traffic.go

```diff
@@ -75,33 +76,38 @@ func (t readWriteSingleKey) Read(ctx context.Context, c *recordingClient, limite
 }
 
 func (t readWriteSingleKey) Write(ctx context.Context, c *recordingClient, limiter *rate.Limiter, key string, newValue string, lm clientId2LeaseIdMapper, cid int, lastValues []*mvccpb.KeyValue) error {
-	putCtx, cancel := context.WithTimeout(ctx, 20*time.Millisecond)
+	writeCtx, cancel := context.WithTimeout(ctx, RequestTimeout)
 
 	var err error
 	switch t.pickWriteOperation() {
 	case Put:
-		err = c.Put(putCtx, key, newValue)
+		err = c.Put(writeCtx, key, newValue)
 	case Delete:
-		err = c.Delete(putCtx, key)
+		err = c.Delete(writeCtx, key)
 	case Txn:
 		var expectValue string
 		if len(lastValues) != 0 {
 			expectValue = string(lastValues[0].Value)
 		}
-		err = c.Txn(putCtx, key, expectValue, newValue)
+		err = c.Txn(writeCtx, key, expectValue, newValue)
 	case PutWithLease:
 		leaseId := lm.LeaseId(cid)
 		if leaseId == 0 {
-			leaseId, err = c.LeaseGrant(ctx, t.leaseTTL)
-			lm.AddLeaseId(cid, leaseId)
+			leaseId, err = c.LeaseGrant(writeCtx, t.leaseTTL)
+			if err != nil {
+				lm.AddLeaseId(cid, leaseId)
+				limiter.Wait(ctx)
+			}
```

Comment:
Do you intentionally ignore the lease when `err == nil`? 

```suggestion
			lm.AddLeaseId(cid, leaseId)
			if err != nil {
				limiter.Wait(ctx)
			}
```

Reply:
Ups, I had wrong condition.